listenbrainz/webserver/login/provider.py

```diff
@@ -33,16 +36,23 @@ def get_user():
     data = s.get('oauth2/userinfo').json()
     musicbrainz_id = data.get('sub')
     musicbrainz_row_id = data.get('metabrainz_user_id')
-    user_exists_check = db_user.get_by_mb_id(musicbrainz_id)
-    user = db_user.get_or_create(musicbrainz_row_id, musicbrainz_id)
-    if user:
-        if user_exists_check is None:
-            ts.set_empty_cache_values_for_user(musicbrainz_id)
-        if not user['musicbrainz_row_id']:
-            db_user.update_musicbrainz_row_id(musicbrainz_id, data['metabrainz_user_id'])
-        return User.from_dbrow(user)
+
+    user = db_user.get_by_mb_row_id(musicbrainz_row_id, musicbrainz_id)
+    user_email = None
+    if mb_engine:
+        user_email = mb_editor.get_editor_by_id(musicbrainz_row_id)['email']
+
+    if user is None:
+        if current_app.config["REJECT_USERS_WITHOUT_EMAIL"] and user_email is None:
+            return None
+        db_user.create(musicbrainz_row_id, musicbrainz_id, email=user_email)
+        user = db_user.get_by_mb_row_id(musicbrainz_row_id, musicbrainz_id)
+        ts.set_empty_cache_values_for_user(musicbrainz_id)
     else:
-        return None
+        user["email"] = user_email
+        db_user.update_user_email(musicbrainz_id, user_email)
```

Comment:
we're missing case here where the user doesn't have an email configured in musicbrainz, but already has an account in LB and signs in again.

Reply:
Its intended that existing users without email be able to login. The warnings are not shown by this part of the code. Its done in https://github.com/metabrainz/listenbrainz-server/pull/1422/files#diff-7271692c5f71146001b6530c09733f46e8fa24e6183b335adbeb7e71428dca7bR31-R39